driver-core/src/main/com/mongodb/internal/operation/CommandBatchCursor.java

```diff
@@ -55,92 +52,65 @@
     private static final Logger LOGGER = Loggers.getLogger("operation");
     private static final FieldNameValidator NO_OP_FIELD_NAME_VALIDATOR = new NoOpFieldNameValidator();
-    private static final String CURSOR = "cursor";
-    private static final String POST_BATCH_RESUME_TOKEN = "postBatchResumeToken";
-    private static final String OPERATION_TIME = "operationTime";
-    private static final String MESSAGE_IF_CLOSED_AS_CURSOR = "Cursor has been closed";
-    private static final String MESSAGE_IF_CLOSED_AS_ITERATOR = "Iterator has been closed";
 
     private final MongoNamespace namespace;
-    private final ServerAddress serverAddress;
     private final int limit;
     private final Decoder<T> decoder;
     private final long maxTimeMS;
     private int batchSize;
+    @Nullable
     private final BsonValue comment;
+    private CommandCursorResult<T> commandCursorResult;
+    @Nullable
     private List<T> nextBatch;
     private int count;
-    private BsonDocument postBatchResumeToken;
-    private BsonTimestamp operationTime;
     private final boolean firstBatchEmpty;
-    private int maxWireVersion = 0;
+    private final int maxWireVersion;
     private final ResourceManager resourceManager;
 
-    CommandBatchCursor(final CommandCursorResult<T> firstCommandCursorResult, final int limit, final int batchSize, final Decoder<T> decoder) {
-        this(firstCommandCursorResult, limit, batchSize, decoder, null, null);
-    }
-
-    CommandBatchCursor(final CommandCursorResult<T> firstCommandCursorResult, final int limit, final int batchSize, final Decoder<T> decoder,
-            @Nullable final BsonValue comment, @Nullable final ConnectionSource connectionSource) {
-        this(firstCommandCursorResult, limit, batchSize, 0, decoder, comment, connectionSource, null, null);
-    }
-
-    CommandBatchCursor(final CommandCursorResult<T> firstCommandCursorResult, final int limit, final int batchSize, final long maxTimeMS,
-            final Decoder<T> decoder, @Nullable final BsonValue comment, @Nullable final ConnectionSource connectionSource,
-            @Nullable final Connection connection) {
-        this(firstCommandCursorResult, limit, batchSize, maxTimeMS, decoder, comment, connectionSource, connection, null);
-    }
-
-    CommandBatchCursor(final CommandCursorResult<T> firstCommandCursorResult, final int limit, final int batchSize, final long maxTimeMS,
-            final Decoder<T> decoder, @Nullable final BsonValue comment, @Nullable final ConnectionSource connectionSource,
-            @Nullable final Connection connection, @Nullable final BsonDocument result) {
-        isTrueArgument("maxTimeMS >= 0", maxTimeMS >= 0);
-        this.maxTimeMS = maxTimeMS;
-        this.namespace = firstCommandCursorResult.getNamespace();
-        this.serverAddress = firstCommandCursorResult.getServerAddress();
+    CommandBatchCursor(
+            final ServerAddress serverAddress,
```

Comment:
Why pass the ServerAddress separately when it can be pulled off either the source or connection (neither of which is nullable)?  That's where all the callers of this method are getting it from anyway.

Reply:
Done